### PR TITLE
Scheduler: Clarify ExecutionModel docs

### PR DESCRIPTION
**Pull request by timbertson**
An attempt to clarify the docs in relation to https://github.com/monix/monix/issues/1222

### CODE REVIEW

File: _docs/3x/execution/scheduler.md [hunk 1]
```diff
@@ -319,13 +323,11 @@ Currently there are 3 execution models available:
...
-  asynchronously on each step, being basically the mode of operation
-  for Scala's `Future`.
+  specifies that each task should be treated as if it were asynchronous,
+  being basically the mode of operation for Scala's `Future`.
 - [SynchronousExecution]({{ page.path | api_base_url }}monix/execution/ExecutionModel$.html#SynchronousExecution)
-  specifies that synchronous execution should always be preferred,
-  for as long as possible, being basically the mode of operation
-  for the Scalaz `Task`.
+  specifies that tasks should be executed synchronously for as long
+  as possible, being basically the mode of operation for the Scalaz `Task`.
```
Comment (Avasil) [inline _docs/3x/execution/scheduler.md:330]: How about we change `Scalaz Task` to `cats.effect.IO` ? It should be more recognizable nowadays and Scalaz Task could be confused with ZIO which uses the same model as Monix Task by default

File: _docs/3x/execution/scheduler.md [hunk 2]
```diff
@@ -300,16 +300,20 @@ testScheduler.execute(runnable1)
 testScheduler.execute(runnable2)
 ```
 
+Note that the Execution model (explained below) still matters for a TestScheduler.
```
Comment (Avasil) [inline _docs/3x/execution/scheduler.md:303]: I feel like it's not strictly about `ExecutionModel` but about a `Task` being possibly synchronous in which case we could change it to be asynchronous and one of the option is to use `AlwaysAsyncExecution` (we could also use executeAsync if we are only bothered about the start of execution).

Maybe we could add a small example with `Task { println("Immediate!") }` + `runToFuture` and provide a solution?

File: _docs/3x/execution/scheduler.md [hunk 3]
```diff
@@ -300,16 +300,20 @@ testScheduler.execute(runnable1)
...
+When using the default `SynchronousExecution`, some tasks will be evaluated
```
Comment (Avasil) [inline _docs/3x/execution/scheduler.md:304]: I think the default is `BatchedExecution`, is it not?

File: _docs/3x/execution/scheduler.md (hunk 1 above)
Comment (timbertson) [reply _docs/3x/execution/scheduler.md]: Done 👍

File: _docs/3x/execution/scheduler.md (hunk 2 above)
Comment (timbertson) [reply _docs/3x/execution/scheduler.md]: True. I've reworded, what do you think?

I wasn't sure what a good example would look like, you're welcome to add to this PR if you have one in mind :)

File: _docs/3x/execution/scheduler.md [hunk 4]
```diff
@@ -300,16 +300,21 @@ testScheduler.execute(runnable1)
 testScheduler.execute(runnable2)
 ```
 
+Note that most methods for spawning a Task (`runToFuture`, etc) will execute work
+on the current thread until an async boundary is encountered. If you want all execution
+to happen on the TestScheduler (in order to manually `tick()` it), you can use
+`.asyncBoundary` on the Task, or set the TestScheduler's ExecutionModel to AlwaysAsyncExecution.
```
Comment (Avasil) [inline _docs/3x/execution/scheduler.md:306]:
```suggestion
`executeAsync` on the Task, or set the TestScheduler's ExecutionModel to AlwaysAsyncExecution.
```

`asyncBoundary` will insert it after the `Task`

**Pull request merged**